crates/shared/src/price_estimation.rs

```diff
@@ -294,8 +331,12 @@ impl Display for Arguments {
         display_option(f, "balancer_sor_url", balancer_sor_url)?;
         display_secret_option(f, "one_inch_spot_price_api_key: {:?}", one_inch_api_key)?;
         writeln!(f, "one_inch_spot_price_api_url: {}", one_inch_url)?;
-        display_secret_option(f, "coin_gecko_api_key: {:?}", coin_gecko_api_key)?;
-        writeln!(f, "coin_gecko_api_url: {}", coin_gecko_url)?;
+        display_secret_option(
+            f,
+            "coin_gecko_api_key: {:?}",
+            &coin_gecko.coin_gecko_api_key,
```

Comment:
I do not like the `coin_gecko.coin_gecko_...` but there was no other way since we `flatten` it, and each field should contain the `coin_gecko_` prefix.